libraries/xcode.rb

```diff
@@ -3,18 +3,20 @@
 
 module MacOS
   class Xcode
-    attr_reader :version
-    attr_reader :intended_path
-    attr_reader :download_url
+    attr_reader :version, :intended_path, :download_url
 
     def initialize(semantic_version, intended_path, download_url = '')
       @semantic_version = semantic_version
       @intended_path = intended_path
       @download_url = download_url
-      @version = if download_url.empty?
-                   latest_xcode_revision(Xcode::Version.new(semantic_version))
+      @version = -> { determine_version }
+    end
+
+    def determine_version
```

Comment:
Is there a test for this newly added logic? It might be less obvious since the method doesn't have any parameters.

Reply:
`determine_version` is just an extraction of the assignment that's from `initialize`. We're covered here with the tests that instantiate Xcode with the `download_url` parameter.